### PR TITLE
Fix error in windows provider when the environment as an existing set of variables name that only differs by case

### LINKED ISSUE #6305: get-childitem env: fails on Windows when an environment variable is defined twice, cased differently
The problem:
`get-childitem env:` fails on Windows when an environment variable exists twice, each instance cased differently.

The scenario:
On Windows, set the environment variable `NPM_CONFIG_PATH` to control where the npm cache is created. And use npm to run unit tests, use powershell within a unit test, and from powershell invoke `get-childitem env:`. Note, it is common to use `npm test` to build and test a nodejs project.

It appears npm internally creates environment variables to manage state, and a lower-case variable `npm_config_path` is created. When powershell is invoked from the unit test, the environment variables `NPM_CONFIG_PATH` and `npm_config_path` are both set.

Steps to reproduce
------------------

On Windows, tested with npm v5.6.0 in my PATH.

```powershell
Write-Host "******************** SETTING NPM_CONFIG_PATH ********************"
$null = mkdir env-test
cd env-test
$env:NPM_CONFIG_PATH = $PWD
Write-Host "******************** CREATING PACKAGE.JSON ********************"
& npm init --force --yes
Write-Host "******************** MODIFYING PACKAGE.JSON ********************"
$package = Get-Content package.json | ConvertFrom-Json
$package.scripts.test = "pwsh -command Get-ChildItem env:"
Set-Content -LiteralPath package.json -Encoding UTF8 -Value (ConvertTo-Json -InputObject $package -Depth 99)
Get-Content package.json
Write-Host "******************** TESTING ENV PS DRIVE, VIA NPM ********************"
& npm test
```

Expected behavior
-----------------

PowerShell should gracefully handle the duplicate environment variable. The `set` builtin in cmd.exe handles it gracefully. .NET `Environment.GetEnvironmentVariables()` handles it gracefully.

Actual behavior
---------------

Get-ChildItem fails with \"An item with the same key has already been added. Key: NPM_CONFIG_PATH\"

Environment data
----------------

```powershell
> $PSVersionTable
Name                           Value
----                           -----
PSVersion                      6.0.1
PSEdition                      Core
GitCommitId                    v6.0.1
OS                             Microsoft Windows 10.0.16299
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### PR DESCRIPTION
**Pull request by TravisEz13**
## PR Summary

fix error in windows provider when the environment as an existing set of variables name that only differs by case
    - make the provider storage for the environment on windows ignore duplicates
    - add tests to verify existing environment get-item behavior

fixes #6305 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: test/powershell/Modules/Microsoft.PowerShell.Management/Get-Item.Tests.ps1
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Get-Item.Tests.ps1]: Please use upper case in `Should`.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Get-Item.Tests.ps1
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Get-Item.Tests.ps1]: Please remove extra line.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Get-Item.Tests.ps1
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Get-Item.Tests.ps1]: Please use upper case in Should.

**Review by iSazonov — COMMENTED** [review]
Leave a comment

**Review by daxian-dbw — DISMISSED** [review]
LGTM. Please address @iSazonov's comments.

**Comment by TravisEz13** [comment]
fix error in windows provider when the environment as an existing set of variables name that only differs by case
- make the provider storage for the environment on windows case sensitive
- This does not make the provider actually case sensitive on windows as the dotnet APIs are not case sensitive
- add tests to verify existing environment get-item behavior

**Comment by TravisEz13** [comment]
@mklement0 has a comment on the fix https://github.com/PowerShell/PowerShell/issues/6305#issuecomment-371656505 . I'll wait until this is resolved to merge

**Review by daxian-dbw — APPROVED** [review]

**Pull request merged**
